### PR TITLE
Create a discovery Graph that uses docker to run node discovery

### PR DESCRIPTION
**Pull request by keedya**
depends on : https://github.com/RackHD/on-http/pull/392
depends on: https://github.com/RackHD/on-imagebuilder/pull/52/files
depends on : https://github.com/RackHD/on-tasks/pull/293

references:  
- https://osclouds.wordpress.com/2015/11/23/announcing-a-new-docker-based-hanlon-microkernel
- https://groups.google.com/forum/#!topic/rackhd/cSYYxhtPWY0

### CODE REVIEW

**Comment by keedya** [comment]
@RackHD/corecommitters @zyoung51 @jlongever @benbp

File: lib/graphs/rancher-discovery-graph.js [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        }
+    },
+    tasks: [
+        {
+            label: 'bootstrap-rancher',
+            taskName: 'Task.Linux.Bootstrap.Rancher'
+        },
+        {
+            label: 'catalog-dmi',
+            taskName: 'Task.Catalog.dmi'
```
Comment (yyscamper) [inline lib/graphs/rancher-discovery-graph.js]: Miss waitOn the success of `bootstrap-rancher`

**Comment by yyscamper** [comment]
:+1:

**Comment by benbp** [comment]
👍

**Comment by zyoung51** [comment]
test this please

**Comment by brianparry** [comment]
test this please

**Pull request merged**